manim/animation/composition.py

```diff
@@ -48,20 +48,18 @@ def get_run_time(self):
         return super().get_run_time()
 
     def begin(self):
-        for anim in self.animations:
-            anim.begin()
-
-    def finish(self):
-        for anim in self.animations:
-            anim.finish()
+        for anim, start_time, end_time in self.anims_with_timings:
+            if np.isclose(start_time, 0):
+                anim.begin()
 
     def clean_up_from_scene(self, scene):
         for anim in self.animations:
             anim.clean_up_from_scene(scene)
 
     def update_mobjects(self, dt):
         for anim in self.animations:
-            anim.update_mobjects(dt)
+            if hasattr(anim, 'starting_mobject'):
+                anim.update_mobjects(dt)
```

Comment:
Question: an animation only has a `starting_object` if its `begin()` method has been called, correct? So this conditional is checking whether the animation is currently running?

If so, we should make note to implement a cleaner way of doing that in the future. Not in this PR tho.

Reply:
From what I've looked at today, I think there are some animations that don't set `starting_object` when their `begin` is called. It might actually be better to refactor this still on this PR: maybe by adding a boolean `has_started` set to `False` on `Animation.__init__`, and to true in all the implementing `begin` methods?

Reply:
Agreed. Suggest `is_running`.